community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -591,46 +588,63 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
 
     override def delete(id: Long) {
       try {
-        transactionalContext.statement.dataWriteOperations().relationshipDelete(id)
+        writes().relationshipDelete(id)
       } catch {
         case _: exceptions.EntityNotFoundException => // node has been deleted by another transaction, oh well...
       }
```

Comment:
In the Kernel API relationshipDelete is idempotent I think, so we can remove the try-catch as well.